bindings/rust/src/lib.rs

```diff
@@ -135,7 +135,29 @@ impl From<f64> for Value {
 }
 
 /// The result of an execution.
-pub type ExecutionResult = sys::FizzyExecutionResult;
+pub struct ExecutionResult(sys::FizzyExecutionResult);
+
+impl ExecutionResult {
+    /// True if execution has resulted in a trap.
+    pub fn trapped(&self) -> bool {
+        self.0.trapped
+    }
+
+    /// The optional return value. Only a single return value is allowed in WebAssembly 1.0.
+    pub fn value(&self) -> Option<Value> {
+        if !self.0.trapped && self.0.has_value {
```

Comment:
Documentation says otherwise, but this is too defensive. It is guaranteed that `has_value` if `false` if `trapped` is `true`.

Reply:
@gumb0 suggested this. Where is it guaranteed?

Reply:
> Documentation says otherwise, but this is too defensive. It is guaranteed that `has_value` if `false` if `trapped` is `true`.

Here it's the opposite - when `trapped` is `false`, we have to check `has_value`.

What I suggested was changing `if has_value` -> `if !trapped && has_value`

Reply:
> What I suggested was changing if has_value -> if !trapped && has_value

This is what the code has now.

Reply:
The `ExecuteResult` constructor guarantees it. If you want to keep it, swap the checks: `has_value && !trapped`.

Reply:
I can also add the trapped as an assertion, which is only present in debug builds.

Reply:
> The ExecuteResult constructor guarantees it.

The C API has no constructor guarantee for it though?

Reply:
> > The ExecuteResult constructor guarantees it.
> 
> The C API has no constructor guarantee for it though?

No constructor guarantees but `FizzyExecutionResult wrap(const fizzy::ExecutionResult&)` just copies all three fields always.

Reply:
> I can also add the trapped as an assertion, which is only present in debug builds:

This variant looks good.

Reply:
I'm ok with changing `trapped` check to assertion.